### ISSUE TITLE
connect() ignores cursor_factory if connection_factory is also specified

### ISSUE THREAD

**Issue by dandanlen** [psycopg/psycopg2#1019]
Hi there, 

I tried setting the `connection_factory` and `cursor_factory` args through the `psycopg2.connect` method, but it looks like the cursor factory isn't being used. As a result, rows are returned as tuples instead of dicts. 

When you use the `cursor_factory` in isolation it works fine: 
```python
with psycopg2.connect(dsn,
                      cursor_factory=DictCursor) as conn:
    with conn.cursor() as cur:
        print(cur.row_factory)
        # prints: <class 'psycopg2.extras.DictRow'>
```

However if you also set a `connection_factory` (Am I using this wrong? 🤷‍♂ ):
```python
with psycopg2.connect(dsn,
                      connection_factory=LoggingConnection,
                      cursor_factory=DictCursor) as conn:
    conn.initialize(logging.Logger('test'))
    with conn.cursor() as cur:
        print(cur.row_factory)
        # prints: None
```

I can work around it by specifying the `cursor_factory` when I create the cursor. 
```python
with psycopg2.connect(dsn, connection_factory=LoggingConnection) as conn:
    conn.initialize(logging.Logger('test'))
    with conn.cursor(cursor_factory=DictCursor) as cur:
        print(cur.row_factory)
        # prints: <class 'psycopg2.extras.DictRow'>
```

I'm using version  2.8.4, Python 3.7.4 on mac.

**Comment by dvarrazzo** [comment]
It's a contrived use case, but a genuine bug, affecting pretty much all the provided connection subclasses.

**Comment by dandanlen** [comment]
Well, it was easy to work around but I wouldn't call it a contrived use case... Anyway, thanks for the quick response!

**Comment by dvarrazzo** [comment]
The fact nobody stumbled in it in 15 years the adapter is used in all the world makes it a pretty unlikely use case.

**Comment by danielfoguelman** [comment]
> The fact nobody stumbled in it in 15 years the adapter is used in all the world makes it a pretty unlikely use case.

I stumbled upon it. Do you think it still is an unlikeley use case?

**Comment by dvarrazzo** [comment]
Considering that the bug was closed 3 years ago, yes, pretty unlikely.

**Comment by danielfoguelman** [comment]
I would say, if you find an unlikely event to happen twice in a finite time interval, it is not as unlikely as you thought. But well, the reporter of this issue was kind enough to provide a way to overcome this issue.

**Comment by dvarrazzo** [comment]
@danielfoguelman I don't understand: this issue was closed in 2019 in 2.8.5. Are you using an older version than that, and complaining, or have you found the same issue in a more recent version?

**Comment by danielfoguelman** [comment]
It is true that I'm using an older version and complaining, Danielle. Sorry about that. From this ticket history I thought that it was closed as a not-fix and it amused me to think that two persons could find an unlikely bug. Now I see you merged the fix in `2.9.5` version.
Again, I didn't ment to be rude, I thought wrong about the whole thing. Have a nice week!
